### ISSUE TITLE
Error in create gallery page 'The option "cascade_validation" does not exist.'

### ISSUE THREAD

**Issue by valiknet18** [sonata-project/SonataMediaBundle#1111]
### Environment

Bundle version:     3.2.0
Symfony version 3.1.3
php version 5.6
## Subject

Hello, after install sonata media bundle, in create gallery page (admin), i had error `The option "cascade_validation" does not exist.`.  I don't find resolve to this problem :C.
## Actual results

Page with error.

![image](https://cloud.githubusercontent.com/assets/5772019/18064482/5037a346-6e38-11e6-9a82-0836e1c5e9b3.png)

Stack trace:

```
Stack Trace
in vendor/symfony/symfony/src/Symfony/Component/OptionsResolver/OptionsResolver.php at line 685   -
            ksort($clone->defined);
            ksort($diff);
            throw new UndefinedOptionsException(sprintf(
                (count($diff) > 1 ? 'The options "%s" do not exist.' : 'The option "%s" does not exist.').' Defined options are: "%s".',
                implode('", "', array_keys($diff)),
                implode('", "', array_keys($clone->defined))
at OptionsResolver ->resolve (array('sonata_field_description' => object(FieldDescription), 'type' => 'sonata_type_admin', 'modifiable' => true, 'type_options' => array('sonata_field_description' => object(FieldDescription), 'data_class' => 'Application\Sonata\MediaBundle\Entity\GalleryHasMedia'), 'cascade_validation' => true, 'label_render' => false, 'label' => 'form.label_gallery_has_medias')) 
in vendor/symfony/symfony/src/Symfony/Component/Form/ResolvedFormType.php at line 95   + 
at ResolvedFormType ->createBuilder (object(FormFactory), 'galleryHasMedias', array('sonata_field_description' => object(FieldDescription), 'type' => 'sonata_type_admin', 'modifiable' => true, 'type_options' => array('sonata_field_description' => object(FieldDescription), 'data_class' => 'Application\Sonata\MediaBundle\Entity\GalleryHasMedia'), 'cascade_validation' => true, 'label_render' => false, 'label' => 'form.label_gallery_has_medias')) 
in vendor/symfony/symfony/src/Symfony/Component/Form/Extension/DataCollector/Proxy/ResolvedTypeDataCollectorProxy.php at line 81   + 
at ResolvedTypeDataCollectorProxy ->createBuilder (object(FormFactory), 'galleryHasMedias', array('sonata_field_description' => object(FieldDescription), 'type' => 'sonata_type_admin', 'modifiable' => true, 'type_options' => array('sonata_field_description' => object(FieldDescription), 'data_class' => 'Application\Sonata\MediaBundle\Entity\GalleryHasMedia'), 'cascade_validation' => true, 'label_render' => false, 'label' => 'form.label_gallery_has_medias')) 
in vendor/symfony/symfony/src/Symfony/Component/Form/FormFactory.php at line 85   + 
at FormFactory ->createNamedBuilder ('galleryHasMedias', 'sonata_type_collection', null, array('sonata_field_description' => object(FieldDescription), 'type' => 'sonata_type_admin', 'modifiable' => true, 'type_options' => array('sonata_field_description' => object(FieldDescription), 'data_class' => 'Application\Sonata\MediaBundle\Entity\GalleryHasMedia'), 'cascade_validation' => true, 'label_render' => false, 'label' => 'form.label_gallery_has_medias')) 
in vendor/symfony/symfony/src/Symfony/Component/Form/FormBuilder.php at line 106   + 
at FormBuilder ->create ('galleryHasMedias', 'sonata_type_collection', array('sonata_field_description' => object(FieldDescription), 'type' => 'sonata_type_admin', 'modifiable' => true, 'type_options' => array('sonata_field_description' => object(FieldDescription), 'data_class' => 'Application\Sonata\MediaBundle\Entity\GalleryHasMedia'), 'cascade_validation' => true, 'label_render' => false, 'label' => 'form.label_gallery_has_medias')) 
in vendor/symfony/symfony/src/Symfony/Component/Form/FormBuilder.php at line 269   + 
at FormBuilder ->resolveChildren () 
in vendor/symfony/symfony/src/Symfony/Component/Form/FormBuilder.php at line 215   + 
at FormBuilder ->getForm () 
in var/cache/dev/classes.php at line 11361   + 
at AbstractAdmin ->buildForm () 
in var/cache/dev/classes.php at line 10434   + 
at AbstractAdmin ->getForm () 
in vendor/sonata-project/admin-bundle/Controller/CRUDController.php at line 502   + 
at CRUDController ->createAction () 
```

**Comment by greg0ire** [comment]
Please don't post screenshot of the error page, the only interesting thing in it is the message, we don't really care about the little ghost :) Copy/paste the message, it will make your post more SEO-friendly

**Comment by greg0ire** [comment]
Here the first occurence of `cascade_validation` in your stack trace 

> at FormBuilder ->create ('galleryHasMedias', 'sonata_type_collection', array('sonata_field_description' => object(FieldDescription), 'type' => 'sonata_type_admin', 'modifiable' => true, 'type_options' => array('sonata_field_description' => object(FieldDescription), 'data_class' => 'Application\Sonata\MediaBundle\Entity\GalleryHasMedia'), 'cascade_validation' => true, 'label_render' => false, 'label' => 'form.label_gallery_has_medias')) 
in vendor/symfony/symfony/src/Symfony/Component/Form/FormBuilder.php at line 269

**Comment by valiknet18** [comment]
Sorry :(
Thanks, it's help me, but i get new error `You have requested a non-existent service "request". Did you mean one of these: "monolog.logger.request", "request_stack", "assetic.request_listener", "router.request_context", "validate_request_listener", "data_collector.request"?` in create media page.

**Comment by greg0ire** [comment]
It will be hard to help without a stack trace…

**Comment by valiknet18** [comment]
Sorry.
`
Stack Trace
in vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php at line 266   -
                        }
                    }
                    throw new ServiceNotFoundException($id, null, null, $alternatives);
                }
                return;
at Container ->get ('request') 
in vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php at line 389   + 
at Controller ->get ('request') 
in vendor/sonata-project/media-bundle/Controller/MediaAdminController.php at line 32   + 
at MediaAdminController ->createAction (object(Request)) 
at call_user_func_array (array(object(MediaAdminController), 'createAction'), array(object(Request))) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 153   + 
at HttpKernel ->handleRaw (object(Request), '1') 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 68   + 
at HttpKernel ->handle (object(Request), '1', true) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php at line 169   + 
at Kernel ->handle (object(Request)) 
in web/app_dev.php at line 30   + `

**Comment by greg0ire** [comment]
Can you please format it like I did in your first messagE?

**Comment by valiknet18** [comment]
## Subject

I get error 'You have requested a non-existent service "request". Did you mean one of these: "monolog.logger.request", "request_stack", "assetic.request_listener", "router.request_context", "validate_request_listener", "data_collector.request"? in create media page (admin).

Stack trace:

```
Stack Trace
in vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php at line 266   -
                        }
                    }
                    throw new ServiceNotFoundException($id, null, null, $alternatives);
                }
                return;
at Container ->get ('request') 
in vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php at line 389   + 
at Controller ->get ('request') 
in vendor/sonata-project/media-bundle/Controller/MediaAdminController.php at line 32   + 
at MediaAdminController ->createAction (object(Request)) 
at call_user_func_array (array(object(MediaAdminController), 'createAction'), array(object(Request))) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 153   + 
at HttpKernel ->handleRaw (object(Request), '1') 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php at line 68   + 
at HttpKernel ->handle (object(Request), '1', true) 
in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php at line 169   + 
at Kernel ->handle (object(Request)) 
in web/app_dev.php at line 30   + 
```

**Comment by greg0ire** [comment]
The problem probably comes from [this line](https://github.com/sonata-project/SonataMediaBundle/blob/3.x/Controller/MediaAdminController.php#L32). Can you try replacing `$this->get('request')` with $request?

**Comment by valiknet18** [comment]
Yes, thanks, it's helped to resolve my problem.

**Comment by greg0ire** [comment]
Great! Can you submit PRs for that?

**Comment by valiknet18** [comment]
Yes)

**Comment by jordisala1991** [comment]
This is fixed on latest release of SonataMediaBundle 3.4.0
